web/src/compose_actions.js

```diff
@@ -32,6 +31,23 @@ import * as stream_bar from "./stream_bar";
 import * as stream_data from "./stream_data";
 import * as unread_ops from "./unread_ops";
 
+let compose_clear_box_hooks = [];
+let compose_cancel_hooks = [];
+
+export function register_compose_box_clear_hooks(hooks) {
+    compose_clear_box_hooks = [...compose_clear_box_hooks, ...hooks];
```

Comment:
This unnecessarily takes linear time in the number of hooks already registered; many calls would take quadratic time. I know we don’t call this often, but `compose_clear_box_hooks.push(...hooks)` is simpler *and* more efficient. Or better, just accept one hook at a time: why register `[f, g, h]` when you could register `() => {f(); g(); h()}`.

Reply:
Yeah makes sense!